dsl/src/Quickstrom/Spec.purs

```diff
@@ -30,38 +32,46 @@ type Path
 
 -- | A possible action to generate. Quickstrom uses action values when searching the
 -- | DOM for possible actions to generate and perform.
-data Action
+data BaseAction
   = Focus Selector
   | KeyPress Char
   | EnterText String
   | Click Selector
+  | Await Selector
+  | AwaitSecs (Tuple Int Selector)
```

Comment:
I like the idea and considered it at the time but I ended up going with a discrete number of seconds due to the way `awaitElement` is written. I would propose that long term this is the right solution but it would have to be accompanied with a rewrite of `awaitElement`. At this time, I didn't have the cycles (and it's not my focus) to refactor/rewrite `awaitElement` to avoid the "drift" inherent with it and support arbitrary durations.

Reply:
OK, yeah let's push this for another PR.